packages/veritone-react-common/src/components/SearchBar/TranscriptModal/index.js

```diff
@@ -8,4 +17,117 @@ const getTranscriptLabel = modalState => {
   };
 };
 
-export { getTranscriptLabel };
+
+export default class TranscriptSearchModal extends React.Component {
+  static propTypes = {
+    open: bool,
+    modalState: shape({
+      search: string,
+      language: string
+    }),
+    cancel: func
+  };
+  static defaultProps = {
+    modalState: {
+      search: '',
+      language: 'en'
+    },
+    cancel: () => console.log('You clicked cancel')
+  };
+
+  state = {
+    filterValue: null || this.props.modalState.search
+  };
+
+  onChange = event => {
+    this.setState({
+      filterValue: event.target.value
+    });
+  };
+
+  returnValue() {
+    if (!this.state.filterValue || this.state.filterValue.trim().length === 0) {
+      return;
+    } else {
+      return {
+        search: this.state.filterValue ? this.state.filterValue.trim() : null,
+        language: 'en'
+      };
+    }
+  }
+
+  render() {
+    return (
+      <TranscriptSearchForm
+        cancel={this.props.cancel}
+        defaultValue={
+          (this.props.modalState && this.props.modalState.search) || ''
+        }
+        onChange={this.onChange}
+        inputValue={this.state.filterValue}
+      />
+    );
+  }
+}
+
+export const TranscriptSearchForm = ({
+  defaultValue,
+  cancel,
+  onChange,
+  onKeyPress,
+  inputValue
+}) => {
+  return (
+    <TextField
+      id="transcript_search_field"
+      autoFocus
+      margin="none"
+      defaultValue={defaultValue}
+      onChange={onChange}
+      onKeyPress={onKeyPress}
+      placeholder="Phrase to search"
+      fullWidth
+    />
+  );
+};
+
+TranscriptSearchModal.propTypes = {
+  defaultValue : string,
+  cancel: func,
+  onChange: func,
+  onKeyPress: func,
+  inputValue: string
+}
+
+TranscriptSearchModal.defaultProps = {
+  modalState: { search: '', language: 'en' }
+};
+
+const TranscriptConditionGenerator = modalState => {
+  return {
+    operator: 'query_string',
+    field: 'transcript.transcript',
+    value: (modalState.search && modalState.search.toLowerCase()) || ''
+  };
+};
+
+const TranscriptDisplay = modalState => {
```

Comment:
You can remove this. This is the unclean version of the getTranscriptLabel function.